docs/src/modules/components/AppSearch.js

```diff
@@ -119,16 +119,7 @@ export default function AppSearch() {
   const userLanguage = useSelector((state) => state.options.userLanguage);
   const t = useSelector((state) => state.options.t);
 
-  React.useEffect(() => {
-    const styleNode = loadCSS(
-      'https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css',
-      document.querySelector('#app-search'),
-    );
-
-    return () => {
-      styleNode.parentElement.removeChild(styleNode);
-    };
-  }, []);
+  useLazyCSS('https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css', '#app-search');
```

Comment:
If we take the case of a page transition between `/components/speed-dial/` -> `/components/rating/` we remove the CSS of Algolia to append them back a few ms later (it doesn't happen for the other assets). We could wonder:

Should we be worried about removing and appending CSS a few ms later?

Reply:
> Should we be worried about removing and appending CSS a few ms later?

What are you worried about?

Reply:
It's more my curiosity speaking than any of my fears. I'm curious to better understand how the CSS engine performs.

I imagine the concern is close to: What's wrong with leaking a few CSS files? I assume it boils down to improving the performance of the page. I don't have enough knowledge of how the CSS engine performs to tell. I thought that it was interesting to raise the above side effect of page transition, it's not obvious when looking at the code.

Reply:
> What's wrong with leaking a few CSS files?

We're not leaking anything here. We're actually leaking less. The docsearch css is reloaded on master as well.

Reply:
I just realized: This change didn't even change anything for algolia. Any behavior you see related to docsearch.min.css happens on master as well.

For the prism css it's more of a precaution in case we ever unmount it.